Kerpecheva/datagrid grouping updates (#810)
telerik/maui-docs: controls/datagrid/grouping/group-header-template.md

---
title: Group Header Template
page_title: .NET MAUI DataGrid Documentation - GroupHeaderTemplate
description: "Review the Telerik .NET MAUI DataGrid Property Group Descriptor documentation article to learn more about the property grouping function you can use."
position: 5
slug: datagrid-group-header-template
---

# Group Header Template for .NET MAUI DataGrid

> .NET MAUI DataGrid provides the ability to fully customize its group headers in order to achieve the desired look when the data is grouped. Through the `GroupHeaderTemplate` property of the DataGrid you can set a custom DataTemplate to the group headers.

Reviewer: ```suggestion
The .NET MAUI DataGrid features fully customizable group headers that allow you to tweak the appearance and behavior of the headers used to group data within the grid. You can achieve this customization by using the `GroupHeaderTemplate` property of the DataGrid, which allows you to set a custom `DataTemplate` to the group headers.
```

The BindingContext of the `GroupHeaderTemplate` is a `GroupHeaderContext` object and it includes the following properties:

> * Descriptor: Specifies the used descriptor for the grouping;

Reviewer: Style all properties in this list in `monospace`.

> * Descriptor: Specifies the used descriptor for the grouping;

Reviewer: ```suggestion
* `Descriptor`: Specifies the descriptor used for the grouping.
```

* Group: Gets details on the group such as:
    * Items: Gets the child items of the group.

>     * Key: Gets the specific for the group key.

Reviewer: ```suggestion
    * `Key`: Gets the key specific for the group.
```

* IsExpanded: Defines a value indicating whether the group is currently expanded (has its child items visible).
* Level: Gets the zero-based level (or the depth) of the group.

> The following example demonstrates how to apply a sample GroupHeaderTemplate to the DataGrid:

Reviewer: ```suggestion
The following example demonstrates how to apply a sample `GroupHeaderTemplate` to the DataGrid:
```

**1.** Add the DataGrid definition to the page with a sample `GroupHeaderTemplate`:

<snippet id='datagrid-groupheader-template-xaml' />

> **2.** Add the ViewModel class:

Reviewer: ```suggestion
**2.** Add the `ViewModel` class:
```

<snippet id='datagrid-grouping-viewmodel' />

**3.** Add the data item used for binding the DataGrid:

<snippet id='datagrid-grouping-object' />

> **4.** Set the binding context of the DataGrid to the ViewModel class:

Reviewer: ```suggestion
**4.** Set the binding context of the DataGrid to the `ViewModel` class:
```
